--- a/arch/arm64/boot/dts/rockchip/rk3588.dtsi	2024-08-16 22:19:17.702563000 +0800
+++ b/arch/arm64/boot/dts/rockchip/rk3588.dtsi	2024-08-17 11:09:27.073691831 +0800
@@ -437,6 +437,16 @@
 		rockchip,usbdpphy-grf = <&usbdpphy1_grf>;
 		rockchip,vo-grf = <&vo0_grf>;
 		status = "disabled";
+		
+		usbdp_phy1_dp: dp-port {
+			#phy-cells = <0>;
+			status = "disabled";
+		};
+
+		usbdp_phy1_u3: u3-port {
+			#phy-cells = <0>;
+			status = "disabled";
+		};
 	};
 
 	combphy1_ps: phy@fee10000 {
